Public/Src/Cache/DistributedCache.Host/Service/Internal/DistributedContentStoreFactory.cs

```diff
@@ -320,43 +321,37 @@ private async Task ApplySecretSettingsForLlsAsync(RedisContentLocationStoreConfi
                 value => eventStoreConfiguration.MaxEventProcessingConcurrency = value);
         }
 
-        private AzureBlobStorageCredentials[] GetStorageCredentials(Dictionary<string, string> settings, StringBuilder errorBuilder)
+        private AzureBlobStorageCredentials[] GetStorageCredentials(Dictionary<string, Credentials> secrets, StringBuilder errorBuilder)
         {
             var storageSecretNames = GetAzureStorageSecretNames(errorBuilder);
-            if (storageSecretNames == null)
-            {
-                return null;
-            }
-
-            return storageSecretNames.Select(name => {
-                return new AzureBlobStorageCredentials(GetRequiredSecret(settings, name));
-            }).ToArray();
+            return storageSecretNames?.Select(name => GetRequiredSecret(secrets, name) as AzureBlobStorageCredentials).ToArray();

```

Comment:
do you need to filter out null here?

Reply:
I think there is actually an operator for that, let me check...

---
In reply to: [303156470](https://github.com/microsoft/BuildXL/pull/587#discussion_r303156470) [](ancestors = 303156470)

Reply:
[https://stackoverflow.com/questions/4015930/when-to-use-cast-and-oftype-in-linq](https://stackoverflow.com/questions/4015930/when-to-use-cast-and-oftype-in-linq)

---
In reply to: [303156549](https://github.com/microsoft/BuildXL/pull/587#discussion_r303156549) [](ancestors = 303156549,303156470)